misc.c

```diff
@@ -3063,3 +3063,42 @@ lib_contains_symbol(const char *path, const char *s)
 	return ret;
 #endif /* HAVE_NLIST_H */
 }
+
+ssize_t
+write_msg(const char *str, int flags)
```

Comment:
I think you should use one of the utf8.c `*mprintf()` functions here, otherwise you are writing arbitrary characters from a (usually fine but potentially untrustworthy) server to the user's tty. The `logit()` calls that you're replacing do filter these using `strnvis()`, but as you observe this is too restrictive wrt UTF8

Reply:
So this can be done with those indeed, but they look like too complex for this purpose, so I used another behavior but let me know if you prefer using *mprintf funcs.